runtime/manta-pc/src/lib.rs

```diff
@@ -500,23 +508,29 @@ match_type! {
 pub type Barrier = (
 	TakeWeightCredit,
 	AllowTopLevelPaidExecutionFrom<All<MultiLocation>>,
-	AllowUnpaidExecutionFrom<ParentOrParentsExecutivePlurality>,
-	// ^^^ Parent and its exec plurality get free execution
+	AllowUnpaidExecutionFrom<All<MultiLocation>>,
 );
 
 pub struct XcmConfig;
 impl Config for XcmConfig {
 	type Call = Call;
 	type XcmSender = XcmRouter;
 	// How to withdraw and deposit an asset.
-	type AssetTransactor = LocalAssetTransactor;
+	type AssetTransactor = manta_xcm_support::MantaTransactorAdaptor<
+		Balances,
+		MantaXassets,
+		LocationToAccountId,
+		AccountId,
+		CurrencyId,
+		manta_xcm_support::MultiLocationToCurrencyId<MultiLocationMapCurrencyId>,
+	>;
 	type OriginConverter = XcmOriginToTransactDispatchOrigin;
-	type IsReserve = NativeAsset;
+	type IsReserve = manta_xcm_support::TrustedParachains<TrustedChains>;
 	type IsTeleporter = NativeAsset; // <- should be enough to allow teleportation of DOT
 	type LocationInverter = LocationInverter<Ancestry>;
 	type Barrier = Barrier;
 	type Weigher = FixedWeightBounds<UnitWeightCost, Call>;
-	type Trader = UsingComponents<IdentityFee<Balance>, DotLocation, AccountId, Balances, ()>;
+	type Trader = UsingComponents<IdentityFee<Balance>, AcalaLocation, AccountId, Balances, ()>;
```

Comment:
It seems it has to be hardcoded. Look at the definition of `UsingComponents`:
```rust
pub struct UsingComponents<
	WeightToFee: WeightToFeePolynomial<Balance = Currency::Balance>,
	AssetId: Get<MultiLocation>,
	AccountId,
	Currency: CurrencyT<AccountId>,
	OnUnbalanced: OnUnbalancedT<Currency::NegativeImbalance>,
>(
	Weight,
	Currency::Balance,
	PhantomData<(WeightToFee, AssetId, AccountId, Currency, OnUnbalanced)>,
);
```
`AssetId` only supports single value instead of `Vec<MultiLocation>` or something else.

We might post this issue to polkadot.

Reply:
I just went through related xmc code from polkadot, I think we need to implement our own 'UsingComponents', just impl this trait `WeightTrader `.
I'll create an issue for this.